cmd/contour/serve.go

```diff
@@ -213,6 +218,29 @@ func (ctx *serveContext) tlsconfig() *tls.Config {
 	}
 }
 
+// verifytlsflags indicates if the TLS flags are set up correctly.
+// This needs to handle four cases:
+// 1. Flags indicate no TLS required, and are configured correctly. Returns false and nil.
+// 2. Flags indicate no TLS required, but no `--insecure` present. Returns false and error.
+// 2. Flags indicate TLS required, and are not configured correctly. Returns false and error.
+// 3. Flags indicate TLS required, and are configured correctly. Returns true and nil.
+func (ctx *serveContext) verifytlsflags() (bool, error) {
```

Comment:
nit, i know i'm not very consistent with this with helper funcs, but please rename this to `verifyTLSFlags`

https://github.com/golang/go/wiki/CodeReviewComments#initialisms